provider/workspace_owner.go

```diff
@@ -53,7 +53,16 @@ func workspaceOwnerDataSource() *schema.Resource {
 			_ = rd.Set("session_token", os.Getenv("CODER_WORKSPACE_OWNER_SESSION_TOKEN"))
 			_ = rd.Set("oidc_access_token", os.Getenv("CODER_WORKSPACE_OWNER_OIDC_ACCESS_TOKEN"))
 
-			return nil
+			if os.Getenv("CODER_WORKSPACE_OWNER_LOGIN_TYPE") == "" {
+				diags := req.Config.Get(ctx, &rd)
+				diags = append(diags, diag.Diagnostic{
```

Comment:
The `diags` variable you define above is not in the scope of the `return` statement on line 65, so this will not compile.

Instead, you can define a new `diags diag.Diagnostics` variable at the start of the function and reference it here.

Reply:
Thanks and done!

I was not sure if creating a new `diag.Diagnostics` will remove the previous warnings.